build.gradle

```diff
@@ -301,7 +301,7 @@ String bwcFilePath = "src/test/resources/org/opensearch/ad/bwc/"
     testClusters {
         "${baseName}$i" {
             testDistribution = "ARCHIVE"
-            versions = ["7.10.2","1.2.0-SNAPSHOT"]
+            versions = ["7.10.2","1.3.0-SNAPSHOT"]
```

Comment:
We should also test compatibility between minor versions.
Like 1.2.0 -> 1.3.0. 
cc: @VachaShah

Reply:
Definitely agree that we should now test with multiple upgrade paths. One of the ways it is done in the OpenSearch repo is there is a list of bwcVersions that are iterated over and their upgrade paths are tested with respect to the current version.

Reply:
@VachaShah if I'm understanding, you are saying that we should support BWC for the following scenarios, right?

7.10 -> 1.3
1.0 -> 1.3
1.1 -> 1.3
1.2 -> 1.3

I briefly looked at OpenSearch's `build.gradle`, seems it's quite different. Do you know of any plugin that's done this already? I guess it would mostly involve adding more loops to iterate over the older versions. Not sure if there's a cleaner solution.

Overall, agreed that more BWC testing could be supported. Rather than blocking this PR how about I open a tracking issue?

Reply:
@ohltyler absolutely agreed. As long as we take care of this before 1.3 releases we should be good.
Sooner the better :)

Reply:
@ohltyler Yeah thats correct about the scenarios. I will see if any plugin has done this already.